packages/helpers-gatsby/src/Tags.js

```diff
@@ -36,37 +36,54 @@ const Tags = (props) => {
         visibility: props.visibility,
     }
 
+    let keyIndex = 0
+    const generateKey = (pre) => {
+        keyIndex = keyIndex + 1
+        return `${pre}_${keyIndex}`
+    }
+
     if (props.separator) {
-        opts.separator = React.isValidElement(props.separator) ? props.separator :
-            <><span className={props.separatorClasses}>{props.separator}</span></>
+        if (React.isValidElement(props.separator)) {
+            opts.separator = props.separator
+        } else {
+            Object.defineProperty(opts, 'separator', {
+                get() {
+                    return (
+                        <span className={props.separatorClasses} key={generateKey('separator')}>
```

Comment:
This is cool! I cracked my 🧠when I tried to solve this 👏

Reply:
Key generator for `separator` is really cool! 😄 The only possible issue I see there is when developer passes `separator` as a prop, and the condition here - `React.isValidElement(props.separator)` is fulfilled. That will lead to React element for separator being passed as-is, and if developer didn't pass a `key` to that element, it will have the unique key error. 

A good idea to solve for it and not rely on developer input is wrapping the `props.separator` component into another which has the generated key. And while React Fragments (`<>`) are not needed here normally as @AileenCGN said since there is already a top level `span`, they can be useful in this scenario as dummy wrapper element for key, as their long-form is [allowed](https://reactjs.org/docs/fragments.html#keyed-fragments) to have `key` as the only attribute.

```
<React.Fragment key={generateKey('separator')}>
	{props.separator}
</React.Fragment>
```

Reply:
Yeah I did wonder about that but didn't know if there was a convention in React around ensuring components that are passed in as props have their own unique keys. Sounds like there isn't so this is a good workaround, thanks @rishabhgrg!